test/gold_email_input_test.dart

```diff
@@ -0,0 +1,147 @@
+@TestOn('browser')
+library polymer_elements.test.gold_email_input;
+
+import 'package:polymer_elements/gold_email_input.dart';
+import 'package:web_components/web_components.dart';
+import 'package:test/test.dart';
+import 'common.dart';
+
+main() async {
+  await initWebComponents();
+
+  group('basic', () {
+    test('invalid input shows error', () {
+      GoldEmailInput input = fixture('basic');
+      input.value = '1234';
+      input.jsElement.callMethod('_onInput');
+      forceXIfStamp(input);
+
+      var container = input.querySelector('paper-input-container');
+      expect(container, isNotNull);
+      expect(container.invalid, isTrue);
+
+      var error = input.querySelector('paper-input-error');
+      expect(error, isNotNull);
+      expect(error.getComputedStyle().display, isNot('none'),
+          reason: 'error is not display:none');
+    });
+
+    test('valid input does not show error', () {
+      GoldEmailInput input = fixture('basic');
+      input.value = 'batman@gotham.org';
+      input.jsElement.callMethod('_onInput');
+      forceXIfStamp(input);
+
+      var container = input.querySelector('paper-input-container');
+      expect(container, isNotNull);
+      expect(container.invalid, isFalse);
+
+      var error = input.querySelector('paper-input-error');
+      expect(error, isNotNull);
+      expect(error.getComputedStyle().display, equals('none'),
+          reason: 'error should be display:none');
+    }, skip: 'https://github.com/dart-lang/polymer_elements/issues/3');
+
+    test('empty required input shows error', () {
+      GoldEmailInput input = fixture('basic');
+      forceXIfStamp(input);
+
+      var error = input.querySelector('paper-input-error');
+      expect(error, isNotNull);
+      expect(error.getComputedStyle().display, isNot('none'),
+          reason: 'error is not display:none');
+    });
+  });
+
+  group('a11y', () {
+    test('has aria-labelledby', () {
+      GoldEmailInput input = fixture('basic');
+      expect(input.inputElement.getAttribute('aria-labelledby'), isNotNull);
+      expect(input.inputElement.getAttribute('aria-labelledby'),
+          input.querySelector('label').id,
+          reason: 'aria-labelledby points to the label');
+    });
+  });
+
+  void testEmail(address, valid) {
+    GoldEmailInput input = fixture('basic');
+    forceXIfStamp(input);
+
+    var container = input.querySelector('paper-input-container');
+    expect(container, isNotNull);
+
+    input.value = address;
+    input.jsElement.callMethod('_onInput');
+    var errorString = address + ' should be ' + (valid ? 'valid' : 'invalid');
```

Comment:
lets switch this to string interpolation as well `'$address should be ${valid ? 'valid' : 'invalid'}'`